src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,141 @@
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Encoder;
+
+public class IntakeSubsystem extends SubsystemBase {
+  private TalonFX m_moveIntakeArm; // motor of arm
+  private TalonFX m_spinIntake; // motor of intake
+  private Encoder m_Encoder;
+  private DigitalInput m_sensor; 
+  private VelocityVoltage spinRequest1;
+  private double intakeMotorVelocity; 
+  private double intakeMotorAcceleration; 
+  
+  // int lowerIntakeId: Id for lowerng motors for the intake
+  // int spinIntakeId: Id for spining first intake motor 
+  // initialized the encoder 
+  // String can: String ID of canivore  
+  public IntakeSubsystem(int lowerIntakeId, int spinIntakeId, int channel1, int channel2, int channel3, String can)  {
+    m_moveIntakeArm = new TalonFX(lowerIntakeId, can); 
+    m_spinIntake = new TalonFX(spinIntakeId, can);
+    m_Encoder = new Encoder(channel1, channel2);
+    m_sensor = new DigitalInput(channel3);
+    this.intakeMotorVelocity = Constants.IntakeConstants.INTAKE_MOTOR_VELOCITY;
+    this.intakeMotorAcceleration = Constants.IntakeConstants.INTAKE_MOTOR_ACCELERATION;
+    
+    TalonFXConfiguration configs = new TalonFXConfiguration();
+
+    /* Voltage-based velocity requires a feed forward to account for the back-emf of the motor */
+    configs.Slot0.kP = 0.11; // An error of 1 rotation per second results in 2V output
+    configs.Slot0.kI = 0.5; // An error of 1 rotation per second increases output by 0.5V every second
+    configs.Slot0.kD = 0.0001; // A change of 1 rotation per second squared results in 0.01 volts output
+    configs.Slot0.kV = 0.12; // Falcon 500 is a 500kV motor, 500rpm per V = 8.333 rps per V, 1/8.33 = 0.12 volts / Rotation per second
+    // Peak output of 8 volts
+    configs.Voltage.PeakForwardVoltage = 12;
+    configs.Voltage.PeakReverseVoltage = -12;
+    
+    /* Torque-based velocity does not require a feed forward, as torque will accelerate the rotor up to the desired velocity by itself */
+    configs.Slot1.kP = 5; // An error of 1 rotation per second results in 5 amps output
+    configs.Slot1.kI = 0.1; // An error of 1 rotation per second increases output by 0.1 amps every second
+    configs.Slot1.kD = 0.001; // A change of 1000 rotation per second squared results in 1 amp output
+
+    // Peak output of 40 amps
+    configs.TorqueCurrent.PeakForwardTorqueCurrent = 40;
+    configs.TorqueCurrent.PeakReverseTorqueCurrent = -40;
+
+    /* Retry config apply up to 5 times, report if failure */
+    StatusCode status1 = StatusCode.StatusCodeNotInitialized;
+    StatusCode status2 = StatusCode.StatusCodeNotInitialized;
+    for (int i = 0; i < 5; ++i) {
+      status1 = m_moveIntakeArm.getConfigurator().apply(configs);
+      status2 = m_spinIntake.getConfigurator().apply(configs);
+      if (status1.isOK() || status2.isOK()) break;
```

Comment:
should this be `and`.  This code will exit if one worked and the other did not.   Since it tries 5 times, I suspect you intend to try hard on both motors.